File: frontend/src/entities/playlist.ts
export interface PlaylistApi {
	id: number;
	name: string;
	isFavorite: boolean;
	isPublic: boolean;
	songCount: number;
	userId: number;
	username: string;
}

export class Playlist {
	id: number;
	name: string;
	isFavorite: boolean;
	isPublic: boolean;
	songCount: number;
	userId: number;
	username: string;

	constructor(apiObject: PlaylistApi) {
		this.id = apiObject.id;
		this.name = apiObject.name;
		this.isFavorite = apiObject.isFavorite;
		this.isPublic = apiObject.isPublic;
		this.songCount = apiObject.songCount;
		this.userId = apiObject.userId;
		this.username = apiObject.username;
	}
}

File: vibecheck/src/main/java/hr/tvz/vibecheck/api/playlists/entity/Playlist.java
package hr.tvz.vibecheck.api.playlists.entity;

import hr.tvz.vibecheck.api.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "playlists")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Playlist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_playlist")
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "is_favorite", nullable = false)
    private boolean isFavorite;

    @Column(name = "is_public", nullable = false)
    private boolean isPublic;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}
